WordPress/Classes/System/WordPressAppDelegate.swift

```diff
@@ -35,12 +35,10 @@ class WordPressAppDelegate: UIResponder, UIApplicationDelegate {
     private var noticePresenter: NoticePresenter?
     private var bgTask: UIBackgroundTaskIdentifier? = nil
 
+    private let mainContext = ContextManager.shared.mainContext
```

Comment:
I'm thinking this might conflict the way `ContextManager.overrideSharedInstance` is intended to work.  Assigning the context means the overridden context wouldn't be used later.  Maybe we could tweak this to be a computed property instead of a stored property?

Reply:
I think this also has the side-effect of launching Core Data early in the lifecycle of the app before willfinish or didfinish is triggered.

Reply:
Our idea was to initialize the Core Data stack as soon as possible to avoid any concurrency issue.

> Assigning the context means the overridden context wouldn't be used later. Maybe we could tweak this to be a computed property instead of a stored property?

This `mainContext` is used only by the entities in `WordPressAppDelegate`. In fact, we've injected into them to justify the use here — otherwise, it would seem that this is here but not used at all.

We can keep the property here and do not use it to inject tho. And maybe add a comment explaining why is here?

> I think this also has the side-effect of launching Core Data early in the lifecycle of the app before willfinish or didfinish is triggered.

Maybe we can move that to be the first thing in willfinish?

Reply:
Since this has been _controversy_ I reverted here: cb1a526890d548efd73412412967d6ff12154705